mlflow/utils/rest_utils.py

```diff
@@ -133,7 +134,13 @@ def http_request(
 
     from mlflow.tracking.request_header.registry import resolve_request_headers
 
-    headers = dict({**_DEFAULT_HEADERS, **resolve_request_headers()})
+    request_headers = resolve_request_headers()
+    if _USER_AGENT in request_headers:
+        _DEFAULT_HEADERS[_USER_AGENT] = "{} {}".format(
```

Comment:
I don't think we should mutate `_DEFAULT_HEADERS` here. Otherwise, subsequent requests will duplicate `User-Agent` header values from previous requests.

In fact, it may be simpler to reason about if we create a `DefaultHeaderProvider` class that provides the `User-Agent` header and register it. Then we can remove this special-case code. Thoughts?

Reply:
Thanks @dbczumar I had the same thought but was thinking if we can do that in a different PR. But let me go ahead and make that change in this PR and close it out. Will update the PR with this change.

Reply:
Hi @dbczumar I have update the PR to include a new `DefaultRequestHeaderProvider`. Please review and share your feedback.